### ISSUE TITLE
[BUG] int annotation now sets the type to exact Python int

### ISSUE THREAD

**Issue by da-woods** [cython/cython#4944]
**Describe the bug**

Since 31d40c8c62acef9509675155fe5b5bb8e48dba5a an annotation of `int` now corresponds to an exact Python `int`. This matters on Python 2 because it rejects `long`. Was this intended?

**To Reproduce**
Code to reproduce the behaviour:
```cython
def f(x: int):
  return x+1
```

then from python 2
```python
import testmod
testmod.f(1)  # works
testmod.f(long(1))  # fails
```

**Environment (please complete the following information):**
 - Python version: 2.7
 - Cython version: current master branch

**Additional context**
Obviously this is the long-term plan, but I'm not sure we meant to do it now?

Also, whether or not it's supported we probably need some test coverage

**Comment by scoder** [comment]
We should revert this behaviour at least for Cython 3.0. It'll be ok for Cython 3.1 (which drops Py2 support).

We might get away with changing the type check to `isinstance(x, (int, long))` for Py2, although it's not clear if this has any negative effects on user code.

To mitigate this a little for Cython 3.0, we could accept both types in Py2 only if the `language_level` of the compiled module is 3, and otherwise go back to ignoring the type annotation.

**Comment by scoder** [comment]
I'm actually not sure if we should look at the `language_level` here. It sounds surprising if `x: int` did nothing, just because we're looking at Py2 code. Maybe this should only depend on the Py2/Py3 environment? I think `int` really means "integer" in Python, regardless of the value range.

**Comment by scoder** [comment]
OTOH, you could argue that in Py2 code, `x: int` really is ambiguous, whereas in Py3 code it clearly means "any integer" (thus, "int | long" in Py2). Let's keep the language level difference for now and see what users say.

**Comment by da-woods** [comment]
I'd probably be in a favour of the following:

* language_level=3, Python3 -> `int` annotation means Python `int`
* language_level=3, Python2 -> `int` annotation means Python `int` or Python `long`
* language_level=2, Python3 -> `int` annotation means Python `int`
* language_level=2, Python2 -> `int` anntoation means Python `int`; `long` annotation means Python `long`

So the main difference between what you propose and what I propose is that I think we should do *exactly* what people ask us with `language_level=2`. I don't particularly care about the Python2 behaviour too much, but I think the Python3 `int` means `int` is better for `language_level=2`

**Comment by scoder** [comment]
I've merged #5097 as it was. We'll see how users get along with it in the next release.
